content/blog/2024/2024-02-13-apache-http-client-classic-apis.md

```diff
@@ -0,0 +1,776 @@
+---
+authors: [sagaofsilence]
+title: "Classic APIs Offered by Apache HttpClient"
+categories: ["Java"]
+date: 2024-02-11 00:00:00 +1100
+excerpt: "Classic APIs Offered by Apache HttpClient."
+image: images/stock/0077-request-response-1200x628-branded.jpg
+url: apache-http-client-classic-apis
+---
+
+In this article we are going to learn about classic APIs offered by Apache HttpClient APIs. We are going to explore the different ways Apache HttpClient enable developers to send and receive data over the internet in classic (synchronous) mode. From simple GET requests to complex multipart POSTs, we'll cover it all with real-world examples. So get ready to learn to implement HTTP interactions with Apache HttpClient! 
+
+## The "Create a HTTP Client with Apache HttpClient" Series
+
+This article is the third part of a series:
+
+1. [Introduction to Apache HttpClient](/create-a-http-client-with-apache-http-client/)
+2. [Apache HttpClient Configuration](/apache-http-client-config/)
+3. [Classic APIs Offered by Apache HttpClient](/apache-http-client-classic-apis/)
+4. [Async APIs Offered by Apache HttpClient](/apache-http-client-async-apis/)
+5. [Reactive APIs Offered by Apache HttpClient](/apache-http-client-reactive-apis/)
+
+{{% github "https://github.com/thombergs/code-examples/tree/master/create-a-http-client-wth-apache-http-client" %}}
+
+\
+Let's now learn how to use Apache HttpClient for web communication. We have grouped the examples under following categories of APIs: classic, async and reactive. In this article we will learn about the classic APIs offered by Apache HttpClient.
+
+
+{{% info title="Reqres Fake Data CRUD API" %}}
+We are going to use [Reqres API Server](https://reqres.in) to test different HTTP methods. It is a free online API that can be used for testing and prototyping. It provides a variety of endpoints that can be used to test different HTTP methods. The reqres API is a good choice
+ for testing CORS because it supports all of the HTTP methods that are allowed by CORS.
+{{% /info %}}
+
+## HttpClient (Classic APIs)
+In this section of examples we are going to learn how to use `HttpClient` for sending requests and consuming responses in synchronous mode. The client code will wait until it receives response from the server.
```

Comment:
We could divide this section into two subsection to have a clearer structure.

Reply:
I rearranged the subsections based on CRUD ops. I think it looks better now.